[WIP] Argo CD getting started guide. (#30615)
openshift/openshift-docs: modules/gitops-verifying-argo-cd-self-healing-behavior.adoc

[id="verifying-argo-cd-self-healing-behavior_{context}"]

> = Verifying Argo CD self-healing behavior

Reviewer: I suggest making this module part of this assembly: https://deploy-preview-30615--osdocs.netlify.app/openshift-enterprise/latest/cicd/gitops/configuring-argo-cd-to-recursively-sync-a-git-repository-with-your-application.html

Author: I think it makes sense to add it to both asseblies.

> Argo CD constantly monitors the state of deployed applications, detects differences between the specified manifests in Git and live changes in cluster, and automatically corrects them. This behavior is referred to as self-healing. 

Reviewer: the cluster

> Argo CD constantly monitors the state of deployed applications, detects differences between the specified manifests in Git, and live changes in the cluster, and then automatically corrects them. This behavior is referred to as self-healing. 

Reviewer: Remove comma - in Git, and live - as it is not part of the serial comma

> This section describes the steps to test and observe the self-healing behavior.

Reviewer 2: ```suggestion
You can test and observe self-healing behavior in Argo CD.
```

Reviewer 2: I recommend making this an abstract, located just below the title.

.Prerequisites

> * The sample application was deployed and configured.

Reviewer: was/is

.Procedure

. In the Argo CD dashboard, verify that your application has the `Synced` status.

> . Click the `app-spring-petclinic` tile in the Argo CD dashboard to view the application resources that are deployed to the cluster.

Reviewer: If we use the `app-spring-petclinic` tile we will need to ensure the application created was named as `app-spring-petclinic`. So we will need to add it to the prerequisite. 
`synchronizing-your-application-application-with-your-git-repository.adoc` will need this prereq too. Unfortunately without links since these are modules.

> . In the OpenShift Web Console, navigate to the Developer perspective.

Reviewer: Developer - bold
Web Console - lower case w and c on line 22 too

. Modify the Spring PetClinic deployment and commit the changes to the `app/` directory of the Git repository. Argo CD will automatically deploy the changes to the cluster.

. Test the self-healing behavior by modifying the deployment on the cluster and scaling it up to two pods while watching the application in the OpenShift Web Console.
+
.. Run the following command to modify the deployment:
+
[source,terminal]
----
$ oc scale deployment spring-petclinic --replicas 2  -n spring-petclinic
----
.. In the OpenShift Web Console, notice that the deployment scales up to two pods and immediately scales down again to one pod. Argo CD detected a difference from the Git repository and auto-healed the application on the OpenShift cluster. 

> . In the Argo CD dashboard, click the `app-spring-petclinic` name → *App Details* → *Events*. See the event details of Argo CD detecting that the deployment resources are out of sync on the cluster and resyncing the Git repository to correct it.

Reviewer: name/tile?
click the `app-spring-petclinic` name, and then navigate to *App Details* → *Events*. The *Events* tab displays the event details of Argo CD detecting out of sync deployment resources on the cluster and...

> . In the Argo CD dashboard, click the *app-spring-petclinic* tile → *App Details* → *Events*. The *Events* tab displays Argo CD displays the following events: Argo CD detecting out of sync deployment resources on the cluster and then resyncing the Git repository to correct it.

Reviewer: Remove - Argo CD displays

[id="verifying-argo-cd-self-healing-behavior_{context}"]
= Verifying Argo CD self-healing behavior

> Argo CD constantly monitors the state of the deployed applications, detects differences between the specified manifests in Git and live changes in cluster, and automatically corrects them. This behavior is referred to as self-healing. 

Reviewer 3: I think there are two "the's" too many and one too little. "Argo CD constantly monitors the state of deployed applications, detects differences between the specified manifests in Git and live changes in the cluster, and automatically corrects them."

> You can learn more about self-healing in link:https://argo-cd.readthedocs.io/en/stable/user-guide/auto_sync/#automatic-self-healing[Argo CD documentation].

Reviewer 3: I would specify: "You can learn more about Argo CD self-healing in the"

This section describes the steps to test and observe the self-healing behavior.

.Prerequisites

> * The application was deployed and configured.

Reviewer 3: I would specify. The Argo CD application

.Procedure

> . In the Argo CD dashboard, verify that application has the `Synced` status.

Reviewer 3: your or the application

. Click the `app-spring-petclinic` tile in the Argo CD dashboard to view the application resources that are deployed to the cluster.

. In the OpenShift Web Console, navigate to the Developer perspective.

. Modify the Spring PetClinic deployment and commit the changes to the `app/` directory of the Git repository. Argo CD will automatically deploy the changes to the cluster.

> . Test the self-healing behavior by modifying the deployment on the cluster and scaling it up to two pods while watching the application in the OpenShift Web Console:

Reviewer 3: When I read an instruction followed by a colon I expect to see the step I need to do. If the next sentence also ends in a colon in my opinion that is confusing. Maybe the first sentence could end in a full stop?

+

> .. Run the command to modify the deployment:

Reviewer 3: the following command?

+
[source,terminal]
----
$ oc scale deployment spring-petclinic --replicas 2  -n spring-petclinic
----
.. In the OpenShift Web Console, notice that the deployment scales up to two pods and immediately scales down again to one pod. Argo CD detected a difference from the Git repository and auto-healed the application on the OpenShift cluster. 

> . In Argo CD dashboard, click the `app-spring-petclinic` name → *App Details* → *Events*. See the event details of Argo CD detecting that the deployment resources is out of sync on the cluster and resyncing the Git repository to correct it.

Reviewer 3: In the dashboard

Reviewer 3: resources "are" out of sync, because of plural
